packages/common/src/failure.ts

```diff
@@ -172,7 +195,8 @@ export class ApplicationFailure extends TemporalFailure {
     public readonly nonRetryable?: boolean | undefined | null,
     public readonly details?: unknown[] | undefined | null,
     cause?: Error,
-    public readonly nextRetryDelay?: Duration | undefined | null
+    public readonly nextRetryDelay?: Duration | undefined | null,
+    public readonly category?: ApplicationFailureCategory | undefined | null
```

Comment:
`ApplicationFailure` may also be thrown from Workflows. Does `category` being set has any impact anywhere? If not, do we plan to do so in the near future? Should we mention some limitation in doctypes?

Reply:
It's initially targeted just for activity failure logging, with the idea to expand on this based on user feedback. Aside from that, it's just part of the proto failure.

For the doctypes, do you mean to explicitly call out that the BENIGN category only impacts activity failure logging? I suppose we could, but not sure it's necessary.

Reply:
If the `category` property is only significant for `ApplicationFailure`s thrown from activities, it would be nice to users to indicate that fact. Otherwise, we can expect that some users will spend time trying to understand why their benign `ApplicationFailure` thrown from a workflow are still being logged, and then reporting that as a bug.

Reply:
Updated the doc string:

```
/** 
 * A category to describe the severity and change the observability behavior of an application failure.
 * 
 * Currently, observability behaviour changes are limited to:
 * - activities that fail due to a BENIGN application failure emit DEBUG level logs and do not record metrics
 * 
 * @experimental Category is a new feature and may be subject to change.
 */
 ```